### ISSUE TITLE
the range of loss when training

### ISSUE THREAD

**Issue by laolongboy** [MathiasGruber/PConv-Keras#5]
I want to know the range of loss when training, thx

**Comment by liuguilin1225** [comment]
For my original implementation, the input image and target image are normalized to be in the range of [-1,1]; for the input for VGG based loss, it is also in that range. Thus, the loss won't be too big. For my case, it is between [0, 3].

**Comment by dianxin556** [comment]
why my training loss is so big ,that is begined with about 2000000...

**Comment by liuguilin1225** [comment]
@dianxin556  the L1 loss should be size-averaged.

**Comment by ezavarygin** [comment]
@liuguilin1225 In the paper, you used a VGG model trained on pytorch which requires the image pre-processing you point to on your [website](http://masc.cs.gmu.edu/wiki/partialconv), namely [this](https://github.com/pytorch/examples/blob/master/fast_neural_style/neural_style/utils.py#L29):
1. dividing by 255,
2. subtracting a per-channel mean,
3. dividing by a per-channel std.

It does not seem to lead to the [-1,1] range but rather to [-2.12, 2.25], [-2.04, 2.43] and [-1.80, 2.64] for the R, G and B channels.
Am I missing something or did you use another pre-processing?

**Comment by liuguilin1225** [comment]
@ezavarygin sorry for the late reply.
I mainly followed the link you posted, I guess that is how the original VGG model was trained.  https://github.com/pytorch/examples/blob/master/imagenet/main.py#L198

It doesn't necessarily to be in [-1, 1]. Especially for computing perceptual loss and style loss, we just need to make sure the two compared images are using the same preprocessing scheme.

**Comment by MathiasGruber** [comment]
I've ported the VGG16 model from pytorch to keras now, and also updated the PConv2d layer to use local normalization. Loss ranges should be significantly different now, eh?

**Comment by suvojit-0x55aa** [comment]
I'm facing the same problem while training from scratch the loss starts at [10,9] the model gives brown images for the initial epoch. So I started finetuning with the provided imagenet weights but the loss still seems to start at [10,9] is this expected behaviour ?
Also is the hyperparams of the fixmyphoto.ai same as in the notebook, because i'm getting good results with the website, but with the model I'm not getting similar results.
